src/v/http/client.h

```diff
@@ -299,6 +329,11 @@ auto with_client(client&& cl, Func func) {
 const std::unordered_set<std::variant<boost::beast::http::field, std::string>>
 redacted_fields();
 
+ss::future<boost::beast::http::status>
+status(client::response_stream_ref response);
+
+ss::future<iobuf> drain(client::response_stream_ref response);
```

Comment:
Are these useful to be exposed? Seems like `request_and_collect_response` should be preferred.

Reply:
There are a few other places in the tree where similar functions are duplicated, one example is https://github.com/redpanda-data/redpanda/blob/dev/src/v/cloud_roles/request_response_helpers.cc#L25-L51 - I wanted to use these helpers in those places, not in this PR but in a future cleanup attempt.

It is possible those places could also directly use the `request_and_collect_response` helper but I haven't looked closely there.